data/slack-jackson-dto/src/main/kotlin/io/olaph/slack/dto/jackson/group/im/History.kt

```diff
@@ -0,0 +1,64 @@
+package io.olaph.slack.dto.jackson.group.im
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import io.olaph.slack.dto.jackson.JacksonDataClass
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME,
+        include = JsonTypeInfo.As.PROPERTY,
+        property = "ok",
+        visible = true)
+@JsonSubTypes(
+        JsonSubTypes.Type(value = SuccessfulImHistoryResponse::class, name = "true"),
+        JsonSubTypes.Type(value = ErrorImHistoryResponse::class, name = "false")
+)
+
+@JacksonDataClass
+sealed class SlackImHistoryResponse constructor(@JsonProperty(value = "ok") open val ok: Boolean)
+
+@JacksonDataClass
+data class SuccessfulImHistoryResponse constructor(override val ok: Boolean,
+                                                   val latest: String,
+                                                   val messages: List<Message>,
+                                                   val hasMore: Boolean,
+                                                   val isLimited: Boolean? = false)
+    : SlackImHistoryResponse(ok) {
+    companion object {}
+
+    data class Message(
+            @JsonProperty("type") val type: String,
+            @JsonProperty("ts") val ts: String,
+            @JsonProperty("user") val user: String? = null,
+            @JsonProperty("text") val text: String? = null,
+            @JsonProperty("is_starred") val isStarred: Boolean? = false) {
+        companion object
+    }
+}
+
+@JacksonDataClass
+data class ErrorImHistoryResponse constructor(override val ok: Boolean,
+                                              val error: String)
+    : SlackImHistoryResponse(ok) {
+    companion object
+}
+
+data class SlackImHistoryRequest(private val channel: String,
+                                 private val count: Int? = 100,
+                                 private val inclusive: Boolean? = false,
+                                 private val latest: String? = "now",
+                                 private val oldest: String? = "0",
+                                 private val unreads: Boolean? = false) {
+    companion object {}
+
+    fun toRequestMap(): MutableMap<String, String> {
+        val requestMap = mutableMapOf<String, String>()
+        channel?.let { requestMap.put("channel", it) }
```

Comment:
channel is not nullable, so you can simply add the values to the map without ?.let